### PR TITLE
Deprecate bundle:controller:action and service:method notation

### PR DESCRIPTION
**Pull request by Tobion**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #25910
| License       | MIT
| Doc PR        | 

The `a::b` notation had some awkward limitations. It supported `MyControllerClass::method` where `MyControllerClass` is either plain class or a service with the same name but the class must exists. This meant it did NOT support `my_service_controller_id::method` because the `class_exists` check would fail at the wrong point in time. But it did support services where class name == id, i.e. the new auto registration based psr naming. This made it very confusing.

I enhanced the `a::b` notation to be very straight forward:
- if `a` exists as a service then use `a` as a service
- otherwise try to use `a` as a class, i.e. `new $a()`
- otherwise check if a::b is a static method (only relevant when the class is abstract or has private contructor). this was potentially supported when using array controller syntax. it now works the same when using the `::` string syntax, like in php itself. since it only happens when nothing else works, it does not have any performance impact.

The old `a:b` syntax is deprecated and just forwards to `a::b` now internally, just as bundle:controller:action.
In general I was able to refactor the logic quite a bit because it always goes through `instantiateController` now.
Spotting deprecated usages is very easy as all outdated routing configs will trigger a deprecation with the DelegatingLoader and it will be normalized in the dumped routes. So you don't get a deprecation again in the ControllerResolver. But if the controller does not come from routing, e.g. twigs render controller function, then it will still be triggered there.

- [x] deprecate `a:b:c`
- [x] deprecate `a:b`
- [x] update existing references to `a::b`
- [x] fix tests
- [x] fix/add support for static controllers
- [x] add support for closures as controllers
- [x] update Symfony\Component\Routing\Loader\ObjectRouteLoader
- [x] deprecate \Symfony\Bundle\FrameworkBundle\Controller\ControllerNameParser but we still need to use it in several places for BC.
- [x] add changelog/upgrade
- [x] update controller.service_arguments logic for double colon controller syntax

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php [hunk 1]
```diff
@@ -38,15 +38,14 @@ protected function createController($controller)
         if (false === strpos($controller, '::') && 2 === substr_count($controller, ':')) {
             // controller in the a:b:c notation then
             $controller = $this->parser->parse($controller);
-        }
-
-        $resolvedController = parent::createController($controller);
 
-        if (1 === substr_count($controller, ':') && is_array($resolvedController)) {
-            $resolvedController[0] = $this->configureController($resolvedController[0]);
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: This did not configure the controller when it's invokable. It's fixed now since it's also going through instantiateController now like all the other cases.

File: src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php [hunk 2]
```diff
@@ -33,58 +32,17 @@ public function __construct(ContainerInterface $container, LoggerInterface $logg
...
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getController(Request $request)
-    {
-        $controller = parent::getController($request);
-
-        if (is_array($controller) && isset($controller[0]) && is_string($controller[0]) && $this->container->has($controller[0])) {
-            $controller[0] = $this->instantiateController($controller[0]);
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php]: This does not belong here. I moved it to the base controller resolver. This is much more consistent and also makes array controller work where the class needs to be instantiated with old-school `new $class`.

File: src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php [hunk 3]
```diff
@@ -47,6 +43,14 @@ public function getController(Request $request)
         }
 
         if (is_array($controller)) {
+            if (isset($controller[0]) && is_string($controller[0])) {
+                $controller[0] = $this->instantiateController($controller[0]);
+            }
+
+            if (!is_callable($controller)) {
+                throw new \InvalidArgumentException(sprintf('The controller for URI "%s" is not callable. %s', $request->getPathInfo(), $this->getControllerError($controller)));
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: this case was missing validation. otherwise it fails in the kernel with a type error at https://github.com/symfony/symfony/blob/89d1b65037172bfb8bc4de1eeb438cd5c8de80da/src/Symfony/Component/HttpKernel/Event/FilterControllerEvent.php#L32

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php [hunk 4]
```diff
@@ -37,16 +37,17 @@ protected function createController($controller)
...
+            @trigger_error(sprintf(
+                'Referencing controllers with %s is deprecated since version 4.1 and will be removed in 5.0. Use %s instead.',
+                $deprecatedNotation,
+                $controller
+            ), E_USER_DEPRECATED);
         }
 
-        return $resolvedController;
+        return parent::createController($controller);
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: The whole class can be deprecated after removing the deprecated code in 5.1. Which means removing this class in 6.0. What about deprecating the class right now. Upgrading could then be done by 1/ removing all old notation usage 2/ switching to the Component class equivalent. WDYT?

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php (hunk 4 above)
Comment (mvrhov) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: You mean 4.1 and 5.0 right :)

Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: @fabpot we still need to class for the `ContainerAwareInterface` logic, see `configureController`. So we can't deprecate it now. We might want to remove ContainerAware logic in the future as well, but that's a different issue.

Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: Indeed, I missed that. It could be deprecated at some point, but definitely not part of this work.

File: src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php [hunk 5]
```diff
@@ -47,23 +43,27 @@ public function getController(Request $request)
         }
 
         if (is_array($controller)) {
+            if (isset($controller[0]) && is_string($controller[0])) {
+                $controller[0] = $this->instantiateController($controller[0]);
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: wouldn't this break support for using static methods as controller ?

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php (hunk 4 above)
Comment (stof) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: Can't we move the ContainerAwareInterface logic to the ContainerControllerResolver of the component ?

File: src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php (hunk 5 above)
Comment (Tobion) [reply src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: Yeah I also thought about this. We don't have tests using static methods and `class::method` already had the problem of not supporting static methods while that would a valid php callable.
But I already have a solution for this in my mind that will support both cases.

Comment (Tobion) [reply src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: implemented

**Comment by Tobion** [comment]
Tests should pass once merged.

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php (hunk 4 above)
Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: I don't think it's worth it because there is also logic for the AbstractController which is part of the framework bundle. So it fits in the framework bundle.

File: src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php [hunk 6]
```diff
@@ -47,23 +43,42 @@ public function getController(Request $request)
...
+                        }
+                    } catch (\ReflectionException $reflectionException) {
+                        throw $e;
+                    }
+
+                    throw $e;
+                }
+            }
+
+            if (!is_callable($controller)) {
```
Comment (lyrixx) [inline src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: What about using http://php.net/manual/en/closure.fromcallable.php to ensure it's callable?

Comment (Tobion) [reply src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: I don't know what you intend. In which way does Closure::fromCallable help here?

Comment (lyrixx) [reply src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: The `Closure::fromCallable` method calls `is_callable` for you.
Then, It "standardize" how we (will) manage callable

Comment (Tobion) [reply src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: Using a closure here provides no benefit and the error it would return for wrong callables is worse than what `getControllerError` does. For example it does not hint that an __invoke method might be missing. https://3v4l.org/fJJNs

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php [hunk 7]
```diff
@@ -41,6 +43,8 @@ public function __construct(KernelInterface $kernel)
      */
     public function parse($controller)
     {
+        @trigger_error(sprintf('The %s class is deprecated since version 4.1 and will be removed in 5.0.', __CLASS__), E_USER_DEPRECATED);
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php]: why not put the trigger on top of the file?

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php [hunk 8]
```diff
@@ -19,6 +19,8 @@
  * (Bundle\BlogBundle\Controller\PostController::indexAction).
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @deprecated since version 4.1, will be removed in 5.0.
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php]: `, to be removed in Symfony 5.0`

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php [hunk 9]
```diff
@@ -86,6 +90,8 @@ public function parse($controller)
      */
     public function build($controller)
     {
+        @trigger_error(sprintf('The %s class is deprecated since version 4.1 and will be removed in 5.0.', __CLASS__), E_USER_DEPRECATED);
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php]: `"%s"` quoted + `Symfony 5.0`

File: src/Symfony/Component/HttpKernel/Controller/ControllerResolverInterface.php [hunk 10]
```diff
@@ -17,8 +17,6 @@
  * A ControllerResolverInterface implementation knows how to determine the
  * controller to execute based on a Request object.
  *
- * It can also determine the arguments to pass to the Controller.
```
Comment (ro0NL) [inline src/Symfony/Component/HttpKernel/Controller/ControllerResolverInterface.php]: to be applied on 4.0?

**Review by ro0NL — COMMENTED** [review]
just some random comments :) :+1: another great step forward.

File: src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php [hunk 11]
```diff
@@ -73,14 +73,36 @@ public function load($resource, $type = null)
...
+                        'Referencing controllers with %s is deprecated since version 4.1 and will be removed in 5.0. Use %s instead.',
+                        $deprecatedNotation,
+                        $controller
+                    ), E_USER_DEPRECATED);
+                } catch (\InvalidArgumentException $e) {
+                    // unable to optimize unknown notation
+                }
+            }
+
+            if (1 === substr_count($controller, ':')) {
```
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: You could use an `elseif` here if the condition is mutually exclusive with the one above. It would prevent evaluating this one when the first is true.

File: src/Symfony/Component/HttpKernel/DependencyInjection/RemoveEmptyControllerArgumentLocatorsPass.php [hunk 12]
```diff
@@ -47,8 +47,7 @@ public function process(ContainerBuilder $container)
             } else {
                 // any methods listed for call-at-instantiation cannot be actions
                 $reason = false;
-                $action = substr(strrchr($controller, ':'), 1);
-                $id = substr($controller, 0, -1 - strlen($action));
+                list($id, $action) = explode('::', $controller);
```
Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/DependencyInjection/RemoveEmptyControllerArgumentLocatorsPass.php]: What about using `[$id, $action]` here?

**Review by HeahDude — APPROVED** [review]
Great!

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php (hunk 7 above)
Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php]: because we still use the class and service internally. but the trigger only matters when it gets executed

File: src/Symfony/Component/HttpKernel/Controller/ControllerResolverInterface.php (hunk 10 above)
Comment (Tobion) [reply src/Symfony/Component/HttpKernel/Controller/ControllerResolverInterface.php]: I guess also for 3.4 but no priority for me

File: src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php (hunk 11 above)
Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: Depends if someone used an extended name parser that returned something else. I would leave it like this for simplicitly.

File: src/Symfony/Component/HttpKernel/DependencyInjection/RemoveEmptyControllerArgumentLocatorsPass.php (hunk 12 above)
Comment (Tobion) [reply src/Symfony/Component/HttpKernel/DependencyInjection/RemoveEmptyControllerArgumentLocatorsPass.php]: We could but I don't think we use that syntax in the code base yet.

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php (hunk 8 above)
Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php]: I normalized all deprecation messages

**Comment by chalasr** [comment]
The `controller_name_converter` and `resolve_controller_name_subscriber` services should be deprecated.

**Comment by Tobion** [comment]
I can't deprecate them because we still need to use them internally.

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php [hunk 13]
```diff
@@ -37,16 +37,17 @@ protected function createController($controller)
     {
         if (false === strpos($controller, '::') && 2 === substr_count($controller, ':')) {
             // controller in the a:b:c notation then
-            $controller = $this->parser->parse($controller);
-        }
-
-        $resolvedController = parent::createController($controller);
+            $deprecatedNotation = $controller;
+            $controller = $this->parser->parse($deprecatedNotation);
```
Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: will give an extra notice about the deprecated class being used, right? I would change ControllerNameParser to trigger from its constructor only (muting the notice using an implicit constructor arg for internal uses)

Comment (chalasr) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: or disable the notice in the method call

Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: done

File: src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md [hunk 14]
```diff
@@ -10,6 +10,9 @@ CHANGELOG
  * Added option in workflow dump command to label graph with a custom label
  * Using a `RouterInterface` that does not implement the `WarmableInterface` is deprecated and will not be supported in Symfony 5.0.
  * The `RequestDataCollector` class has been deprecated and will be removed in Symfony 5.0. Use the `Symfony\Component\HttpKernel\DataCollector\RequestDataCollector` class instead.
+ * Deprecated `bundle:controller:action` syntax to reference controllers. Use `serviceOrFqcn::method` instead where `serviceOrFqcn`
+   is either the service ID or the FQCN of the controller.
+ * Deprecated `Symfony\Bundle\FrameworkBundle\Controller\ControllerNameParser`
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md]: This means that the UPGRADE-5.0 file is missing some information.

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php [hunk 15]
```diff
@@ -37,16 +37,17 @@ protected function createController($controller)
...
+            $deprecatedNotation = $controller;
+            $controller = $this->parser->parse($deprecatedNotation, false);
 
-        if (1 === substr_count($controller, ':') && is_array($resolvedController)) {
-            $resolvedController[0] = $this->configureController($resolvedController[0]);
+            @trigger_error(sprintf(
+                'Referencing controllers with %s is deprecated since Symfony 4.1. Use %s instead.',
+                $deprecatedNotation,
+                $controller
+            ), E_USER_DEPRECATED);
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php]: This should be on one line.

File: src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php [hunk 16]
```diff
@@ -73,14 +73,36 @@ public function load($resource, $type = null)
...
+                $deprecatedNotation = $controller;
+
+                try {
+                    $controller = $this->parser->parse($controller, false);
+
+                    @trigger_error(sprintf(
+                        'Referencing controllers with %s is deprecated since Symfony 4.1. Use %s instead.',
+                        $deprecatedNotation,
+                        $controller
+                    ), E_USER_DEPRECATED);
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: On one line as well.

File: src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php [hunk 17]
```diff
@@ -73,14 +73,36 @@ public function load($resource, $type = null)
...
+                    // unable to optimize unknown notation
+                }
+            }
+
+            if (1 === substr_count($controller, ':')) {
+                $controller = str_replace(':', '::', $controller);
+                @trigger_error(sprintf(
+                    'Referencing controllers with a single colon is deprecated since Symfony 4.1. Use %s instead.',
+                    $controller
+                ), E_USER_DEPRECATED);
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: Same here

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 18]
```diff
@@ -659,7 +659,7 @@ private function addValidationSection(ArrayNodeDefinition $rootNode)
                     ->beforeNormalization()
                         ->ifTrue(function ($v) { return isset($v['strict_email']); })
                         ->then(function ($v) {
-                            @trigger_error('The "framework.validation.strict_email" configuration key has been deprecated in Symfony 4.1. Use the "framework.validation.email_validation_mode" configuration key instead.', E_USER_DEPRECATED);
+                            @trigger_error('The "framework.validation.strict_email" configuration key is deprecated since Symfony 4.1. Use the "framework.validation.email_validation_mode" configuration key instead.', E_USER_DEPRECATED);
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Unrelated changes should really be moved to another PR. This is especially important here as people will have a look at this PR when upgrading.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml [hunk 19]
```diff
@@ -70,7 +70,7 @@
         </service>
 
         <service id="session.save_listener" class="Symfony\Component\HttpKernel\EventListener\SaveSessionListener">
-            <deprecated>The "%service_id%" service is deprecated since Symfony 4.1 and will be removed in 5.0. Use the "session_listener" service instead.</deprecated>
+            <deprecated>The "%service_id%" service is deprecated since Symfony 4.1. Use the "session_listener" service instead.</deprecated>
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml]: This should be part of another PR as not related to this one.

File: src/Symfony/Component/HttpFoundation/File/UploadedFile.php [hunk 20]
```diff
@@ -145,13 +145,13 @@ public function guessClientExtension()
      * It is extracted from the request from which the file has been uploaded.
      * Then it should not be considered as a safe value.
      *
-     * @deprecated since 4.1 will be removed in 5.0 use getSize() instead.
+     * @deprecated since Symfony 4.1, to be removed in 5.0. Use getSize() instead.
```
Comment (fabpot) [inline src/Symfony/Component/HttpFoundation/File/UploadedFile.php]: Same here, should be moved to another PR.

File: src/Symfony/Component/HttpKernel/CHANGELOG.md [hunk 21]
```diff
@@ -6,6 +6,7 @@ CHANGELOG
 
  * added orphaned events support to `EventDataCollector`
  * `ExceptionListener` now logs and collects exceptions at priority `2048` (previously logged at `-128` and collected at `0`)
+ * Deprecated `service:action` syntax with a single colon to reference controllers. Use `service::method` instead.
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/CHANGELOG.md]: should also be part of UPGRADE-5.0.md.

File: src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php [hunk 22]
```diff
@@ -98,22 +56,26 @@ protected function instantiateController($class)
...
+        } catch (\Error $e) {
         }
 
         $this->throwExceptionIfControllerWasRemoved($class, $e);
 
-        throw $e;
+        if ($e instanceof \ArgumentCountError) {
+            throw new \InvalidArgumentException(
+                sprintf('Controller "%s" has required constructor arguments and does not exist in the container. Did you forget to define such a service?', $class), 0, $e
+            );
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php]: one line

File: src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php [hunk 23]
```diff
@@ -98,22 +56,26 @@ protected function instantiateController($class)
...
-     * @param \Exception|\Throwable|null $previous
-     */
-    private function throwExceptionIfControllerWasRemoved($controller, $previous = null)
+    private function throwExceptionIfControllerWasRemoved(string $controller, \Throwable $previous)
     {
         if ($this->container instanceof Container && isset($this->container->getRemovedIds()[$controller])) {
-            throw new \LogicException(sprintf('Controller "%s" cannot be fetched from the container because it is private. Did you forget to tag the service with "controller.service_arguments"?', $controller), 0, $previous);
+            throw new \InvalidArgumentException(
+                sprintf('Controller "%s" cannot be fetched from the container because it is private. Did you forget to tag the service with "controller.service_arguments"?', $controller), 0, $previous
+            );
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php]: one line

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/HttpFoundation/File/UploadedFile.php (hunk 20 above)
Comment (Tobion) [reply src/Symfony/Component/HttpFoundation/File/UploadedFile.php]: Luckily I saw you already did that in #26225

**Comment by Tobion** [comment]
@fabpot fixed your requests

**Review by fabpot — APPROVED** [review]

**Comment by Tobion** [comment]
Can we merge this? I'd prefer to not having to rebase this again.

**Comment by fabpot** [comment]
Thank you @Tobion.

File: src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php [hunk 24]
```diff
@@ -73,14 +73,29 @@ public function load($resource, $type = null)
...
+                    $controller = $this->parser->parse($controller, false);
+
+                    @trigger_error(sprintf('Referencing controllers with %s is deprecated since Symfony 4.1. Use %s instead.', $deprecatedNotation, $controller), E_USER_DEPRECATED);
+                } catch (\InvalidArgumentException $e) {
+                    // unable to optimize unknown notation
+                }
+            }
+
+            if (1 === substr_count($controller, ':')) {
+                $controller = str_replace(':', '::', $controller);
```
Comment (Majkl578) [inline src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: Unfortunately this change is a BC break and breaks custom code relying on single colon notation. (Simply commenting this out fixes the BC break.)

Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: I see. Removing this is an option. Do you want to create a PR?

Comment (Majkl578) [reply src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php]: > Do you want to create a PR?

Not necessarily, I don't have knowledge of Symfony test suite so I'd be happy if you do that, unless you are busy. :+1: 

Btw already reported in #27522.
